src/main/java/se/sundsvall/emailreader/service/EmailScheduler.java

```diff
@@ -0,0 +1,62 @@
+package se.sundsvall.emailreader.service;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import se.sundsvall.emailreader.integration.db.CredentialsRepository;
+import se.sundsvall.emailreader.integration.db.EmailRepository;
+import se.sundsvall.emailreader.integration.ews.EWSIntegration;
+import se.sundsvall.emailreader.service.mapper.EmailMapper;
+import se.sundsvall.emailreader.utility.EncryptionUtility;
+
+@Component
+@EnableScheduling
+public class EmailScheduler {
+
+    private static final Logger log = LoggerFactory.getLogger(EmailScheduler.class);
+
+    private final EWSIntegration ewsIntegration;
+
+    private final EmailRepository emailRepository;
+
+    private final CredentialsRepository credentialsRepository;
+
+    private final EmailMapper emailMapper = new EmailMapper();
+
+    private final EncryptionUtility encryptionUtility;
+
+    public EmailScheduler(final EWSIntegration ewsIntegration, final EmailRepository emailRepository,
+        final CredentialsRepository credentialsRepository, final EncryptionUtility encryptionUtility) {
+
+        this.ewsIntegration = ewsIntegration;
+        this.emailRepository = emailRepository;
+        this.credentialsRepository = credentialsRepository;
+        this.encryptionUtility = encryptionUtility;
+    }
+
+    @Scheduled(initialDelayString = "${scheduled.initial-delay}", fixedRateString = "${scheduled.fixed-rate}")
+    public void scheduleEmailReader() {
+
+        log.info("Starting scheduled email reader");
+        final var result = credentialsRepository.findAll();
+        
+        result.forEach(credential -> {
+
+            final var resulty = ewsIntegration
+                .pageThroughEntireInbox(credential.getDestinationFolder(),
+                    credential.getUsername(), encryptionUtility.decrypt(credential.getPassword()), credential.getDomain());
+
+            log.info("Found {} emails for mailbox {}", resulty.size(), credential.getUsername());
+
+            emailRepository.saveAll(emailMapper.toEmailEntites(resulty, credential.getNamespace(), credential.getMunicipalityId()));
+
```

Comment:
Should there be a try/catch here (in case of DB-exceptions, etc.)?

The catch-clause could notify someone about the error and maybe move the emails back to the inbox? 

As the code is implemented now, we will never know that an exception occured. The emails will be moved away, but probably not stored in DB and not available in SupportManagement?

Reply:
You have a very valid point and I agree we should make sure emails dont get lost in the void. 

Instead of moving the email back I opted for a version where we dont move the emails until its stored safely in the database. Will reduce the amounts of calls to the email-server and make sure we have stored all moved emails.